packages/core/test/unit/lib/markdown-it/highlight/HightlightRule.test.ts

```diff
@@ -0,0 +1,83 @@
+import { HighlightRule, HIGHLIGHT_TYPES } from '../../../../../src/lib/markdown-it/highlight/HighlightRule';
+import { HighlightRuleComponent } from '../../../../../src/lib/markdown-it/highlight/HighlightRuleComponent';
+
+describe('parseAllRules', () => {
+  test('parses multiple rules correctly', () => {
+    const allRules = '3,4-5';
+    const rules = HighlightRule.parseAllRules(allRules, 0, 'line1\nline2\nline3\nline4\nline5');
+    expect(rules).toHaveLength(2);
+    expect(rules[0]).toBeInstanceOf(HighlightRule);
```

Comment:
can we assert something more specific here?

Reply:
Done